project/make/.ensure-webserver

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -e
+
+# Build a Go static web server on top of busybox image
+# and compile it for target daemon
+IMAGE="webserver"
+
+dir="$DEST/webserver"
+mkdir -p "$dir"
+(
+	cd "$dir"
+	cat > main.go <<'EOF'
+		package main
+
+		import (
+		  "log"
+		  "net/http"
+		)
+
+		func main() {
+		  fs := http.FileServer(http.Dir("/static"))
+		  http.Handle("/", fs)
+		  log.Panic(http.ListenAndServe(":80", nil))
+		}
+EOF
+
+	cat > Dockerfile <<'EOF'
+		FROM busybox
+		EXPOSE 80/tcp
+		COPY server .
+		RUN ["chmod", "+x", "server"]
```

Comment:
Why not use `<<-'EOF'` here so you can indent the final `EOF` too?  See https://gist.github.com/tianon/34450ecf1c623725daec#file-generate-dockerfile-sh for an example of what I mean -- leading tabs get consumed with `-`.